compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -385,6 +460,93 @@ fn lockstep_iter_size(
                 _ => LockstepIterSize::Unconstrained,
             }
         }
+        TokenTree::MetaVarExpr(_, ref expr) => {
+            let default_rslt = LockstepIterSize::Unconstrained;
+            let Some(ident) = expr.ident() else { return default_rslt; };
+            let name = MacroRulesNormalizedIdent::new(ident.clone());
+            match lookup_cur_matched(name, interpolations, repeats) {
+                Some(matched) => match matched {
+                    MatchedNonterminal(_) => default_rslt,
+                    MatchedSeq(ref ads) => {
+                        default_rslt.with(LockstepIterSize::Constraint(ads.len(), name))
+                    }
+                },
+                _ => default_rslt,
+            }
+        }
         TokenTree::Token(..) => LockstepIterSize::Unconstrained,
     }
 }
+
+/// Used by meta-variable expressions when an user input is out of the actual declared bounds. For
+/// example, index(999999) in an repetition of only three elements.
+fn out_of_bounds_err<'a>(
+    cx: &ExtCtxt<'a>,
+    max: usize,
+    span: Span,
+    ty: &str,
+) -> DiagnosticBuilder<'a> {
+    cx.struct_span_err(span, &format!("{ty} depth can not be less than or equal to {max}"))
```

Comment:
```suggestion
    cx.struct_span_err(span, &format!("{ty} depth must be less than or equal to {max}"))
```

or am I misunderstanding?

Reply:
Wow... It was totally wrong. Thanks!